### PR TITLE
Add summary of each backend to docs

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

N/A

### Give a brief description for the solution you have provided

One of the main questions we get is what backend people should use, so we should probably have the information in the docs to link to.

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks at tutorial in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/topic_guides/backends.md [hunk 1]
```diff
@@ -12,22 +12,60 @@ tags:
...
-
-Linking larger datasets requires highly computationally intensive calculations, and generates datasets which are too large to be processed on a standard laptop. For these scenarios, we recommend using one of Splink's big data backend - currently Spark or AWS Athena. When these backends are used, the SQL generated by Splink is sent to the chosen backend for execution.
-
 The Splink code you write is almost identical between backends, so it's straightforward to migrate between backends. Often, it's a good idea to start working using DuckDB on a sample of data, because it will produce results very quickly. When you're comfortable with your model, you may wish to migrate to a big data backend to estimate/predict on the full dataset.
 
 ## Choosing a backend
 
+### Considerations when choosing a SQL backend for Splink
+When choosing which backend to use when getting started with Splink, there are another factors to consider:
```
Comment (ThomasHepworth) [inline docs/topic_guides/backends.md:20]:
```suggestion
When choosing which backend to use when getting started with Splink, there are a number of factors to consider:
```

File: docs/topic_guides/backends.md [hunk 2]
```diff
@@ -12,22 +12,60 @@ tags:
...
+
+- the size of the datasets
```
Comment (ThomasHepworth) [inline docs/topic_guides/backends.md:22]:
```suggestion
- the size of the dataset(s)
```

File: docs/topic_guides/backends.md [hunk 3]
```diff
@@ -12,22 +12,60 @@ tags:
...
+- the amount of configuration required
```
Comment (ThomasHepworth) [inline docs/topic_guides/backends.md:23]: This is ambiguous, but you may well address it below and I just haven't got there yet.

File: docs/topic_guides/backends.md [hunk 4]
```diff
@@ -12,22 +12,60 @@ tags:
...
+
+If working with Databricks, the Spark backend is recommended, however as the Splink development team does not have access to a Databricks environment there will be instances where we will be unable to provide support.
+
+See the Spark [deduplication example notebook](../demos/example_simple_pyspark.ipynb) to get a better idea of how Splink works with Spark.
+
+### :simple-amazonaws: Athena
+
+Athena is a big data SQL backend provided on AWS which is great for large datasets (10+ million records). It requires access to a live AWS account and as a persistent database, requires some additional management of the tables created by Splink. Athena has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in string fuzzy matching functionality due to the lack of some string functions in Athena's underlying SQL engine, [Presto](https://prestodb.io/docs/current/). At this time, the Athena backend is not being actively used by the Splink development team so receives minimal levels of support.
```
Comment (ThomasHepworth) [inline docs/topic_guides/backends.md:48]:
```suggestion
Athena is a big data SQL backend provided on AWS which is great for large datasets (10+ million records). It requires access to a live AWS account and as a persistent database, requires some additional management of the tables created by Splink. Athena has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in string fuzzy matching functionality due to the lack of some string functions in Athena's underlying SQL engine, [Presto](https://prestodb.io/docs/current/). At this time, the Athena backend is being used sparingly by the Splink development team so receives minimal levels of support.
```

or you could say "in niche use cases"

File: docs/topic_guides/backends.md [hunk 5]
```diff
@@ -12,22 +12,60 @@ tags:
...
+
+In addition, from a development perspective, the neccessity for an AWS connection makes testing Athena code more difficult, so there may be occassional bugs that would normally be caught by our testing framework.
+
+See the Athena [deduplication example notebook](../demos/athena_deduplicate_50k_synthetic.ipynb) to get a better idea of how Splink works with Athena.
+
+### :simple-sqlite: SQLite
+
+SQLite is similar to DuckDB in that it is, generally, more suited to smaller datasets. While not as performant as DuckDB, SQLite is simple to setup and can even be run directly in a Jupyter notebook. SQLite has reasonable, but not complete, coverage for the functions in the Splink [comparison libraries](../comparison_level_library.md), with gaps in array and date comparisons. String fuzzy matching, while not native to SQLite is available via python UDFs which has some [performance implications](#additional-information-for-specific-backends). SQLite is not actively been used by the Splink team so receives minimal levels of support.
```
Comment (ThomasHepworth) [inline docs/topic_guides/backends.md:56]: The wording, "can even be run directly in a ...", makes it sound as if DuckDB can't do that here

**Review by ThomasHepworth — APPROVED** [review]
This is wonderful, thanks!

Some minor comments and then you're welcome to merge away.

File: docs/topic_guides/backends.md (hunk 5 above)
Comment (RossKen) [reply docs/topic_guides/backends.md]: Updates

**Pull request merged**